scripts/guardian-set-init.sh

```diff
@@ -45,68 +80,89 @@ solNFTBridge=$(jq --raw-output '.chains."1".contracts.nftBridgeEmitterAddress' $
+
 
 # solana token bridge
-sed -i "/REGISTER_SOL_TOKEN_BRIDGE_VAA=/c\REGISTER_SOL_TOKEN_BRIDGE_VAA=$solTokenBridgeVAA" ./ethereum/.env
-echo "REGISTER_SOL_TOKEN_BRIDGE_VAA=$solTokenBridgeVAA" >> ./scripts/.env
+upsert_env_file $ethFile $solTokenBridge $solTokenBridgeVAA
+upsert_env_file $envFile $solTokenBridge $solTokenBridgeVAA
 # solana nft bridge
-sed -i "/REGISTER_SOL_NFT_BRIDGE_VAA=/c\REGISTER_SOL_NFT_BRIDGE_VAA=$solNFTBridgeVAA" ./ethereum/.env
-echo "REGISTER_SOL_NFT_BRIDGE_VAA=$solNFTBridgeVAA" >> ./scripts/.env
+upsert_env_file $ethFile $solNFTBridge $solNFTBridgeVAA
+upsert_env_file $envFile $solNFTBridge $solNFTBridgeVAA
 
 
 # ethereum token bridge
-sed -i "/REGISTER_ETH_TOKEN_BRIDGE_VAA=/c\REGISTER_ETH_TOKEN_BRIDGE_VAA=$ethTokenBridgeVAA" ./ethereum/.env
-echo "REGISTER_ETH_TOKEN_BRIDGE_VAA=$ethTokenBridgeVAA" >> ./scripts/.env
+upsert_env_file $ethFile $ethTokenBridge $ethTokenBridgeVAA
+upsert_env_file $envFile $ethTokenBridge $ethTokenBridgeVAA
 # ethereum nft bridge
-sed -i "/REGISTER_ETH_NFT_BRIDGE_VAA=/c\REGISTER_ETH_NFT_BRIDGE_VAA=$ethNFTBridgeVAA" ./ethereum/.env
-echo "REGISTER_ETH_NFT_BRIDGE_VAA=$ethNFTBridgeVAA" >> ./scripts/.env
+upsert_env_file $ethFile $ethNFTBridge $ethNFTBridgeVAA
+upsert_env_file $envFile $ethNFTBridge $ethNFTBridgeVAA
 
 
 # terra token bridge
-sed -i "/REGISTER_TERRA_TOKEN_BRIDGE_VAA=/c\REGISTER_TERRA_TOKEN_BRIDGE_VAA=$terraTokenBridgeVAA" ./ethereum/.env
-echo "REGISTER_TERRA_TOKEN_BRIDGE_VAA=$terraTokenBridgeVAA" >> ./scripts/.env
+upsert_env_file $ethFile $terraTokenBridge $terraTokenBridgeVAA
+upsert_env_file $envFile $terraTokenBridge $terraTokenBridgeVAA
 # terra nft bridge
-sed -i "/REGISTER_TERRA_NFT_BRIDGE_VAA=/c\REGISTER_TERRA_NFT_BRIDGE_VAA=$terraNFTBridgeVAA" ./ethereum/.env
-echo "REGISTER_TERRA_NFT_BRIDGE_VAA=$terraNFTBridgeVAA" >> ./scripts/.env
+upsert_env_file $ethFile $terraNFTBridge $terraNFTBridgeVAA
+upsert_env_file $envFile $terraNFTBridge $terraNFTBridgeVAA
 
 
 # bsc token bridge
-sed -i "/REGISTER_BSC_TOKEN_BRIDGE_VAA=/c\REGISTER_BSC_TOKEN_BRIDGE_VAA=$bscTokenBridgeVAA" ./ethereum/.env
-echo "REGISTER_BSC_TOKEN_BRIDGE_VAA=$bscTokenBridgeVAA" >> ./scripts/.env
-
+upsert_env_file $ethFile $bscTokenBridge $bscTokenBridgeVAA
+upsert_env_file $envFile $bscTokenBridge $bscTokenBridgeVAA
 
-# copy the local .env file to the solana & terra dirs
-cp ./scripts/.env ./solana
-cp ./scripts/.env ./terra/tools
 
+# 7) copy the local .env file to the solana & terra dirs, if the script is running on the host machine
+# chain dirs will not exist if running in docker for Tilt, only if running locally. check before copying.
+if [[ -d ./ethereum ]]; then
+    echo "copying $ethFile to /etherum/.env"
+    cp $ethFile ./ethereum/.env
+fi
+if [[ -d ./solana ]]; then
+    echo "copying $envFile to /solana.env"
+    cp $envFile ./solana/.env
+fi
+if [[ -d ./terra/tools ]]; then
+    echo "copying $envFile to /terra/tools/.env"
+    cp $envFile ./terra/tools/.env
+fi
```

Comment:
```
for envfile in /ethereum/.env /solana/.env /terra/tools/.env; do
    dirname=$(echo $envfile | dirname)
    if [ -d "$dirname" ]; then
        echo "copying $envFile to $envfile"
        cp $envFile ./$envfile
    fi
done
```

Maybe more magical than you'd like, but this is less code (untested, just typed it in like a monkey).

If you'd rather keep what you have, fix the comment for solana to have the correct path: `/solana.env` --> `/solana/.env`

Reply:
Implemented, thanks.